### PR TITLE
fix: [DHIS2-12473] calculated program rule variables

### PR DESCRIPTION
**Pull request by JoakimSM**
[DHIS2-12473](https://jira.dhis2.org/browse/DHIS2-12473)

Fixed a problem that caused the calculated program rule variables to not populate.

### CODE REVIEW

File: src/core_modules/capture-core-utils/rulesEngine/RulesEngine.js [hunk 1]
```diff
@@ -25,23 +25,22 @@ import { normalizeRuleVariable } from './commonUtils/normalizeRuleVariable';
...
         const { variableType } = variableHash;
         let variableValue = normalizeRuleVariable(data, variableType);
         if (variableValue && isString(variableValue)) {
             variableValue = `'${variableValue}'`;
         }
 
-        variablesHash[variableToAssign] = {
-            ...variablesHash[variableToAssign],
+        variablesHash[variableHashKey] = {
+            ...variablesHash[variableHashKey],
```
Comment (superskip) [inline src/core_modules/capture-core-utils/rulesEngine/RulesEngine.js:43]: Nitpick: use variableHash here instead?

**Review by superskip — APPROVED** [review]
LGTM!

**Review by simonadomnisoru — APPROVED** [review]
LGTM!

**Review by jasminenguyennn — APPROVED** [review]
👍

File: src/core_modules/capture-core-utils/rulesEngine/RulesEngine.js (hunk 1 above)
Comment (JoakimSM) [reply src/core_modules/capture-core-utils/rulesEngine/RulesEngine.js:43]: makes sense 👍

**Pull request merged**